subprojects/core/src/main/java/org/gradle/cache/internal/DefaultCrossBuildInMemoryCacheFactory.java

```diff
@@ -82,22 +85,30 @@ public <V> CrossBuildInMemoryCache<Class<?>, V> newClassMap() {
     }
 
     private abstract static class AbstractCrossBuildInMemoryCache<K, V> implements CrossBuildInMemoryCache<K, V>, BuildSessionLifecycleListener {
-        private final Object lock = new Object();
+        private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
```

Comment:
❔  Is there a benefit of keeping the lock reference itself?
Alternatively, is there a benefit of storing the read and write sub-locks in fields? Is there a performance penalty for calling `readLock()` methods, etc?